src/clj/commiteth/eth/core.clj

```diff
@@ -26,24 +28,30 @@
 (defn auto-gas-price? [] (env :auto-gas-price false))
 (defn offline-signing? [] (env :offline-signing true))
 
+(def web3j-obj (atom nil))
+(def creds-obj (atom nil))
+
 (defn wallet-file-path []
   (env :eth-wallet-file))
 
 (defn wallet-password []
   (env :eth-password))
 
-(defn creds []
-  (let [password  (wallet-password)
-        file-path (wallet-file-path)]
-    (if (and password file-path)
-      (WalletUtils/loadCredentials
-       password
-       file-path)
-      (throw (ex-info "Make sure you provided proper credentials in appropriate resources/config.edn"
-                      {:password password :file-path file-path})))))
-
 (defn create-web3j []
-  (Web3j/build (HttpService. (eth-rpc-url))))
+ (or @web3j-obj
+      (swap! web3j-obj (constantly (Web3j/build (HttpService. (eth-rpc-url)))))))
+
+(defn creds []
+  (or @creds-obj
+      (let [password  (wallet-password)
+            file-path (wallet-file-path)]
+        (if (and password file-path)
+          (swap! creds-obj
+                 (constantly (WalletUtils/loadCredentials
+                               password
+                               file-path)))
+          (throw (ex-info "Make sure you provided proper credentials in appropriate resources/config.edn"
+                          {:password password :file-path file-path}))))))
```

Comment:
memoize or defonce might be interesting alternatives to the atom approach here..

Reply:
Yes indeed! For instance, `memo` together with `memo-clear!` from [core.memoize](https://github.com/clojure/core.memoize). We can sprinkle it around the code for some low-hanging fruit.

Reply:
I was actually thinking of Clojure.core functions memoize and defonce. Another alternative might be treating these objects as 'mount' components